### ISSUE TITLE
Fix Max Repay to account for interest

### ISSUE THREAD

**Issue by mikeghen** [Ricochet-Exchange/ricochet-frontend#270]
The max repayment does not properly account for the interest, this is due to how the frontend uses `getDebtAmount` vs `getVaultRepayAmount` method. You don't want to use `getDebtAmount` so somewhere in the repay flow, the repay amount is calculated using `getDebtAmount` when it should use `getValutRepayAmount`. Below are two references to where this is happening in the code: 

https://github.com/Ricochet-Exchange/ricochet-frontend/blob/main/src/api/ethereum.ts#L397

https://github.com/Ricochet-Exchange/ricochet-frontend/blob/b7ac5f9c89b5de6162a96f2b3cf0dac4ba5b434a/src/store/banks/sagas/makeTransaction.ts#L56

Im not sure where this happens, but the problem with `getDebtAmount` is that it doesn't account for interest. Once you repay, the debt amount is updated. So if you use `getDebtAmount` you only get the principal, then you repay, then the interested is accounted for and your left over with the interest as dust.

To recreate this, borrow some funds, wait 1 full day, and then do the repayment. There will be 1 days worth of interest still outstanding once you repay. 

A related issue is that I don't think the interest even shows in the UI until you make a repayment. 

Long story short, `getVaultRepayAmount` should be used to get the repayment due. `getDebtAmount` only updated with interest when a repayment happens.

**Comment by MrSquiggles13** [comment]
Im on it

**Comment by Seroxdesign** [comment]
This should be grouped with #60 and #208  if you can assign these all to me @josealonso since I've been working on this bug for a week now.

**Comment by josealonso** [comment]
> This should be grouped with #60 and #208 if you can assign these all to me @josealonso since I've been working on this bug for a week now.

These was reported one month ago, so I appreciate you taking care of this.

**Comment by Seroxdesign** [comment]
#543 is also linked to this issue, actively being worked on

**Comment by mikeghen** [comment]
@Seroxdesign is this all done? There was a PR right?
